buildchain/buildchain/salt_tree.py

```diff
@@ -98,12 +99,36 @@ def _show(task: types.Task) -> str:
 
     def _run(self) -> None:
         """Generate the nginx configuration."""
+        parts = self._get_parts()
         with Path(self.targets[0]).open('w', encoding='utf-8') as fp:
-            fp.write(container_registry.CONSTANTS)
+            for part in parts:
+                fp.write(part)
+
+    def _get_parts(self) -> Iterator[str]:
+        """Yield all parts that should go in the generated configuration file.
+        """
+        raise NotImplementedError
```

Comment:
minor: could inherit from `abc.ABC` and use the `@abc.abstractmethod` decorator.